packager/packager.cc

```diff
@@ -1520,31 +1521,15 @@ vector<ast::ParsedFile> Packager::run(core::GlobalState &gs, WorkerPool &workers
                     } else {
                         job = rewritePackagedFile(ctx, move(job));
                     }
-
-                    results.emplace_back(move(job));
                 }
             }
-            if (filesProcessed > 0) {
-                resultq->push(move(results), filesProcessed);
-            }
+
+            barrier.DecrementCount();
         });
-        files.clear();
 
-        {
-            vector<ast::ParsedFile> threadResult;
-            for (auto result = resultq->wait_pop_timed(threadResult, WorkerPool::BLOCK_INTERVAL(), gs.tracer());
-                 !result.done();
-                 result = resultq->wait_pop_timed(threadResult, WorkerPool::BLOCK_INTERVAL(), gs.tracer())) {
-                if (result.gotItem()) {
-                    files.insert(files.end(), make_move_iterator(threadResult.begin()),
-                                 make_move_iterator(threadResult.end()));
-                }
-            }
-        }
+        barrier.Wait();
     }
 
-    fast_sort(files, [](const auto &a, const auto &b) -> bool { return a.file < b.file; });
```

Comment:
You're saying this is no longer required because we return the `files` in the same order as the input, so either it was sorted and stays sorted, or was not sorted but doesn't matter that it's sorted?